.circleci/config.yml

```diff
@@ -157,13 +162,26 @@ commands:
             - start_docker_services:
                 env: SNAPSHOT_CI=1
                 services: testagent << parameters.docker_services >>
-            - run:
-                environment:
-                  DD_TRACE_AGENT_URL: http://localhost:9126
-
-                command: |
-                  mv .riot .ddriot
-                  riot list --hash-only '<<parameters.pattern>>' | shuf | circleci tests split | xargs -n 1 ./scripts/ddtest riot -v run --exitfirst --pass-env -s
+            - when:
+                condition:
+                  << pipeline.parameters.coverage >>
+                steps:
+                  - run:
+                      environment:
+                        DD_TRACE_AGENT_URL: http://localhost:9126
+                      command: |
+                          mv .riot .ddriot
+                          riot list --hash-only '<<parameters.pattern>>' | shuf | circleci tests split | xargs -n 1 ./scripts/ddtest riot -v run --exitfirst --pass-env -s
+            - unless:
```

Comment:
I don't love these branches, because we then have to maintain multiple occurrences of this riot list | split | run command.

I am not sure a better way.

Reply:
This is not ideal either, but circleci config doesn't allow us a better way to simplify branches based on pipeline parameters, especially since we're forced to override existing **static** pytest-coverage settings in our `setup.cfg`. Note that I've done some research on changing that to be dynamic but also found it to be non-trivial. 

The other alternative is to simplify it through a bash command, but that wouldn't also be completely clean either.